Lib/test/test_calendar.py

```diff
@@ -735,12 +735,15 @@ def get_weekday_names(width):
     def test_locale_calendar_long_weekday_names(self):
         names = (datetime.date(2001, 1, i+1).strftime('%A') for i in range(7))
         max_length = max(map(len, names))
-        if max_length <= 9:
-            self.skipTest('weekday names are too short')
+        abbrev_names = (datetime.date(2001, 1, i+1).strftime('%a') for i in range(7))
+        abbrev_max_length = max(map(len, abbrev_names))
+
+        if max_length - abbrev_max_length < 3:
+            self.skipTest('standard and abbreviated weekday names have too similar lengths')
```

Comment:
We should compare not only max length of full and abbreviated names, but also compare them with the old threshold, 9. The test tests that `get_weekday_names(9)` uses abbreviated names instead of truncated full names.

```suggestion
        if max_length <= 9:
            self.skipTest('weekday names are too short')
        if abbrev_max_length >= 9:
            self.skipTest('abbreviated weekday names are too long')
```